src/components/transactions/TransactionTemplate.vue

```diff
@@ -359,6 +373,20 @@ watch(
   { immediate: true }
 )
 
+watch(
+  () => props.transactionStatus,
+  (value) => {
+    if (value === TransactionStatus.PENDING) {
+      removeTransaction(txId.value)
+      return
+    }
+
+    if (value === TransactionStatus.REJECTED || value === TransactionStatus.CONFIRMED) {
+      addTransaction(txId.value, value)
+    }
+  }
+)
```

Comment:
As I see the `addTransaction` is called only when transaction is finalized (REJECTED or CONFIRMED), so why calling `removeTransaction` for PENDING status. Is this scenario is possible?

Reply:
got it